pkg/collector/reconcile/deployment_test.go

```diff
@@ -168,15 +168,14 @@ func TestExpectedDeployments(t *testing.T) {
 		assert.Equal(t, int32(2), *actual.Spec.Replicas)
 	})
 
-	t.Run("should not update target allocator deployment when the container image is not updated", func(t *testing.T) {
+	t.Run("should update target allocator deployment when the container image is updated", func(t *testing.T) {
 		ctx := context.Background()
 		createObjectIfNotExists(t, "test-targetallocator", &expectedTADeploy)
 		orgUID := expectedTADeploy.OwnerReferences[0].UID
 
 		updatedParam, err := newParams("test/test-img", "")
 		assert.NoError(t, err)
-		updatedDeploy := targetallocator.Deployment(updatedParam.Config, logger, param.Instance)
-		*updatedDeploy.Spec.Replicas = int32(3)
```

Comment:
Why do you have removed the replica update? Would it more meaningful if we change the test expected test result to 3?

Reply:
this test actually didn't change at all, the git diff is just getting confused from the new name. I'm going to change the order of the new test to hopefully better reflect that.